addons/mail/static/src/core/common/composer.js

```diff
@@ -186,6 +186,18 @@ export class Composer extends Component {
         });
     }
 
+    get avatarUrl() {
+        const params = {};
+        if (this.thread) {
+            Object.assign(params, {
+                thread_model: this.thread.model,
+                thread_id: this.thread.id,
+                ...this.thread.rpcParams,
+            });
+        }
+        return this.store.self.avatarUrl(params);
+    }
+
```

Comment:
This looks much simpler to me, you don't need the `const params` and the `Object.assign`

```suggestion
    get avatarUrl() {
        if (this.thread) {
        return this.store.self.avatarUrl({
                thread_model: this.thread.model,
                thread_id: this.thread.id,
                ...this.thread.rpcParams,
            });
        }
        return this.store.self.avatarUrl();
    }

```